Userland/Applications/Browser/Tab.cpp

```diff
@@ -79,6 +79,42 @@ void Tab::view_source(const URL& url, const String& source)
     window->show();
 }
 
+void Tab::update_status(Optional<String> override = {}, i32 count_waiting = 0)
+{
+    if (override.has_value()) {
+        m_statusbar->set_text(*override);
+        return;
+    }
+
+    // Page load event has not fired yet
+    if (!m_loaded) {
+        // We have a target URL and no pending. Page was just requested.
+        if (m_navigating_url.has_value() && count_waiting == 0) {
+            // ex: "Loading google.com"
+            m_statusbar->set_text(String::formatted("Loading {}", m_navigating_url->host()));
+        }
+        // We have a target URL and pending loads, Page loading required resources
+        else if (m_navigating_url.has_value()) {
+            // ex: "google.com is waiting on 5 resources"
+            if (count_waiting == 1)
+                m_statusbar->set_text(String::formatted("{} is waiting on 1 resource", m_navigating_url->host()));
+            else
+                m_statusbar->set_text(String::formatted("{} is waiting on {} resources", m_navigating_url->host(), count_waiting));
+        }
+        // No destination URL set. This really shouldn't happen...
+        else {
```

Comment:
Outside of spec comments, please avoid this (imo) abhorrent construct

Reply:
Which part specifically? The branch down a "This really shouldn't happen...." path? or just the poor comment usefulness?

Reply:
Specifically the comment between the tail end of `if` and `else` itself.
```suggestion
        } else {
            // No destination URL set. This really shouldn't happen...
```